### PR TITLE
Adding ODL (Operator Discretization Library).

### PR DESCRIPTION
**Pull request by leftaroundabout**
Checklist
- [x]  Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by Emvlt** [comment]
I confirm that I agree to be listed as maintainer

**Comment by leftaroundabout** [comment]
I confirm that I agree to be listed as maintainer

**Comment by ozanoktem** [comment]
I confirm that I agree to be listed as maintainer.

**Comment by leftaroundabout** [comment]
@conda-forge/help-python, ready for review!

**Comment by leftaroundabout** [comment]
@conda-forge-admin, relint

**Comment by leftaroundabout** [comment]
@conda-forge/help-python, is there any problem that prevents this from being merged? It has been a while since I filed the pull request.

I already had some strange error reports, apparently because the recipe-branch had gone out of sync with the mainline. But after updating that, all checks pass again so I really don't see why the recipe couldn't be made into a feedstock.

File: recipes/odl/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,59 @@
+package:
+    name: odl
+    version: "0.8.0"
```
Comment (xhochy) [inline recipes/odl/meta.yaml:3]: Please template this using Jinja or otherwise the conda-forge bots won't work.

File: recipes/odl/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+        - setuptools >=65.6
+
+    host:
+        - python
+        - setuptools
+        - nomkl  # [not win]
+        - future >=0.16
+        - numpy >=1.19, <1.27
+        - scipy >=1.1
+        - packaging >=17.0
```
Comment (xhochy) [inline recipes/odl/meta.yaml:25]: All these dependencies should not be needed in host

File: recipes/odl/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+source:
+    url: https://github.com/odlgroup/odl/archive/refs/tags/v0.8.0.tar.gz
+    sha256: dec092ce1f152707d821146046d4d1441672ad6c22d28dd46e1f77d79c210a38
+
+build:
+    number: 0
+    script: {{ PYTHON }} -m pip install . -vv
+
+requirements:
+    build:
```
Comment (xhochy) [inline recipes/odl/meta.yaml:14]:
```suggestion
    host:
```

File: recipes/odl/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,59 @@
+package:
+    name: odl
+    version: "0.8.0"
+
+source:
+    url: https://github.com/odlgroup/odl/archive/refs/tags/v0.8.0.tar.gz
+    sha256: dec092ce1f152707d821146046d4d1441672ad6c22d28dd46e1f77d79c210a38
+
+build:
```
Comment (xhochy) [inline recipes/odl/meta.yaml:9]:
```suggestion
build:
  noarch: python
```

File: recipes/odl/meta.yaml [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+        - python
+        - future >=0.16
+        - numpy >=1.19
+        - scipy >=1.1
+        - packaging >=17.0
+        - matplotlib-base >=3.4
+
+test:
+    requires:
+        - nomkl  # [not win]
```
Comment (xhochy) [inline recipes/odl/meta.yaml:37]:
```suggestion
```

**Review by xhochy — CHANGES_REQUESTED** [review]
Please indent with two spaces like all other recipes.

File: recipes/odl/meta.yaml (hunk 5 above)
Comment (leftaroundabout) [reply recipes/odl/meta.yaml]: Done in [6f585a9](https://github.com/conda-forge/staged-recipes/pull/27504/commits/6f585a97e5d82eb7001080118f9f8991647a3033)

File: recipes/odl/meta.yaml (hunk 3 above)
Comment (leftaroundabout) [reply recipes/odl/meta.yaml]: [5d684f8](https://github.com/conda-forge/staged-recipes/pull/27504/commits/5d684f894ae7311cfe8028e99cf520d6df648bb9)

File: recipes/odl/meta.yaml (hunk 2 above)
Comment (leftaroundabout) [reply recipes/odl/meta.yaml]: Removed in [5d684f8](https://github.com/conda-forge/staged-recipes/pull/27504/commits/5d684f894ae7311cfe8028e99cf520d6df648bb9)

File: recipes/odl/meta.yaml (hunk 1 above)
Comment (leftaroundabout) [reply recipes/odl/meta.yaml]: Done in [6cf601f](https://github.com/conda-forge/staged-recipes/pull/27504/commits/6cf601f899fc42f5f6fd70645987e1fe226a1b60)

**Comment by leftaroundabout** [comment]
@xhochy thanks. I have adopted all your suggestions, fixed problems in CI, and all the tests pass.

File: recipes/odl/meta.yaml [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+  sha256: dec092ce1f152707d821146046d4d1441672ad6c22d28dd46e1f77d79c210a38
+
+build:
+  noarch: python
+  number: 0
+  script: {{ PYTHON }} -m pip install . -vv
+
+requirements:
+  host:
+    - python >=3.7
```
Comment (xhochy) [inline recipes/odl/meta.yaml:19]:
```suggestion
    - python >=3.7
    - pip
```

**Review by xhochy — CHANGES_REQUESTED** [review]

**Review by xhochy — APPROVED** [review]

**Comment by leftaroundabout** [comment]
@xhochy great, thanks for the help!

**Pull request merged**